dist/api_data.py

```diff
@@ -1430,6 +1430,10 @@ def __ge__(self, other):
 ]),
 
 'WT_SESSION.begin_transaction' : Method([
+    Config('allow_read_timestamp_before_oldest', 'false', r'''
+        allows the caller to specify a read timestamp less than the oldest timestamp but newer
+        than the pinned timestamp, cannot be set to true while also rounding up the read timestamp.
```

Comment:
Given @milkie's statement that this is only used when setting the read timestamp to an already existing read timestamp should we add that constraint to this as well (both code and description)? Not having that constraint (like today) makes the code simpler.

Reply:
I feel like that over complicates things without a lot of gain, and it makes this configuration feel very use-case specific. I guess I'd rather generalise a bit for simplicity.